src/test/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/unit/service/cas2/PersonUtilsTest.kt

```diff
@@ -0,0 +1,79 @@
+package uk.gov.justice.digital.hmpps.approvedpremisesapi.unit.service.cas2
+
+import io.mockk.every
+import org.assertj.core.api.Assertions
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.assertThrows
+import org.springframework.http.HttpMethod
+import org.springframework.http.HttpStatus
+import uk.gov.justice.digital.hmpps.approvedpremisesapi.client.ClientResult
+import uk.gov.justice.digital.hmpps.approvedpremisesapi.factory.InmateDetailFactory
+import uk.gov.justice.digital.hmpps.approvedpremisesapi.factory.OffenderDetailsSummaryFactory
+import uk.gov.justice.digital.hmpps.approvedpremisesapi.model.PersonInfoResult
+import uk.gov.justice.digital.hmpps.approvedpremisesapi.problem.NotFoundProblem
+import uk.gov.justice.digital.hmpps.approvedpremisesapi.util.cas2.getFullInfoForPersonOrThrow
+
+class PersonUtilsTest : OffenderServiceTest() {
+
+  @Test
+  fun `throws NotFoundProblem when offender is PersonInfoResult-NotFound, status 404`() {
+    val crn = "ABC123"
+    every { mockOffenderDetailsDataSource.getOffenderDetailSummary(crn) } returns ClientResult
```

Comment:
could we be even more precise about what's under test here, and mock the return of the `getInfoForPerson` method in the Offender Service?

Reply:
With the recent changes I think this means doing something like:

```kotlin
   val mockOffenderService = mockk<OffenderService>()

    @Test
    fun `throws NotFoundProblem when offender is PersonInfoResult-NotFound, status 404`() {
      val crn = "ABC123"
      every {mockOffenderService.getInfoForPerson(crn)} returns PersonInfoResult.NotFound(crn)
      assertThrows<NotFoundProblem> { offenderService.getFullInfoForPersonOrThrow(crn) }
    }
```
But this results in a mockk exception, presumably because I am mocking OffenderService and calling OffenderService.getFullInfoForPersonOrThrow on it.  Possibly there might be a way to do this (I had a chat with @Gweaton about this too), but I don't currently know what it is.

OK as we are @Harriethw?